src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js

```diff
@@ -24,12 +26,11 @@ const AnchorRenderer = (props) => {
     const displayName = lodashGet(props.tnode, 'domNode.children[0].data', '');
     const parentStyle = lodashGet(props.tnode, 'parent.styles.nativeTextRet', {});
     const attrHref = htmlAttribs.href || '';
-    const internalNewExpensifyPath = (attrHref.startsWith(CONST.NEW_EXPENSIFY_URL) && attrHref.replace(CONST.NEW_EXPENSIFY_URL, ''))
-        || (attrHref.startsWith(CONST.STAGING_NEW_EXPENSIFY_URL) && attrHref.replace(CONST.STAGING_NEW_EXPENSIFY_URL, ''));
-    const internalExpensifyPath = attrHref.startsWith(CONFIG.EXPENSIFY.EXPENSIFY_URL)
-                                    && !attrHref.startsWith(CONFIG.EXPENSIFY.CONCIERGE_URL)
-                                    && attrHref.replace(CONFIG.EXPENSIFY.EXPENSIFY_URL, '');
-
+    const attrPath = lodashReplace(Url.getURLObject(attrHref).path, '/', '');
+    const internalNewExpensifyPath = (Url.hasSameOrigin(attrHref, CONST.NEW_EXPENSIFY_URL) || Url.hasSameOrigin(attrHref, CONST.STAGING_NEW_EXPENSIFY_URL)) && attrPath;
+    const internalExpensifyPath = Url.hasSameOrigin(attrHref, CONFIG.EXPENSIFY.EXPENSIFY_URL)
+                                    && !attrPath.startsWith(CONFIG.EXPENSIFY.CONCIERGE_PATHNAME)
+                                    && attrPath;
```

Comment:
Can we please update line 46 as well?

Reply:
@parasharrajat  I don't see any concern in line 46, Please help to point out it

Reply:
Oh, that code moved to like 40. I think 
```
attrHref.startsWith(CONFIG.EXPENSIFY.EXPENSIFY_URL)
```
can also be changed to use `Url.hasSameOrigin` right.